File: core/src/main/java/pcb/uwu/unit/derived/termodynamics/PressureUnit.java
package pcb.uwu.unit.derived.termodynamics;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.unit.base.LengthUnit;
import pcb.uwu.unit.base.MassUnit;
import pcb.uwu.unit.base.TimeUnit;
import pcb.uwu.unit.derived.area.AreaUnit;
import pcb.uwu.unit.derived.mechanics.ForceUnit;

public class PressureUnit extends CompositeUnit {

	// region base constructor

	public PressureUnit(LengthUnit lengthUnit, MassUnit massUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(massUnit)
				.minor(lengthUnit)
				.minor(timeUnit, 2));
	}

	// endregion

	// region derived constructors

	public PressureUnit(ForceUnit forceUnit, AreaUnit areaUnit) {
		super(new UnitCounter()
				.major(forceUnit.getUnitCounter())
				.minor(areaUnit.getUnitCounter()));
	}

	// endregion

}

File: core/src/main/java/pcb/uwu/unit/derived/termodynamics/PressureUnit.kt
package pcb.uwu.unit.derived.termodynamics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.MassUnit
import pcb.uwu.unit.base.TimeUnit
import pcb.uwu.unit.derived.fundamental.AreaUnit
import pcb.uwu.unit.derived.mechanics.ForceUnit

open class PressureUnit : CompositeUnit
{
    constructor(lengthUnit: LengthUnit,
                massUnit: MassUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(massUnit)
                        .minor(lengthUnit)
                        .minor(timeUnit, 2))
    {
    }

    constructor(forceUnit: ForceUnit,
                areaUnit: AreaUnit)
            : super(UnitCounter()
                        .major(forceUnit.unitCounter)
                        .minor(areaUnit.unitCounter))
}